tests/MockJsonBody.php

```diff
@@ -0,0 +1,117 @@
+<?php
+namespace Auth0\Tests;
+
+/**
+ * Class MockJsonBody.
+ *
+ * @package Auth0\Tests
+ */
+class MockJsonBody
+{
+
+    /**
+     * Mock API JSON data.
+     *
+     * @var array
+     */
+    protected $data = [];
+
+    /**
+     * Backup data to reset.
+     *
+     * @var array
+     */
+    protected $dataBackup = [];
+
+    /**
+     * MockJsonBody constructor.
+     *
+     * @param string $name Endpoint name, used to load the JSON file.
+     *
+     * @throws \Exception When the JSON data file cannot be loaded.
+     */
+    public function __construct($name)
+    {
+        $file_path = AUTH0_PHP_TEST_JSON_DIR.'management-api--'.$name.'.json';
+
+        if (! file_exists( $file_path )) {
+            throw new \Exception( 'Cannot find file '.$file_path );
+        }
+
+        $json_contents    = file_get_contents( $file_path );
+        $this->data       = json_decode( $json_contents, true );
+        $this->dataBackup = $this->data;
+    }
+
+    /**
```

Comment:
Reading from a JSON file is fine. What's not good is having the data being manipulated (filtered, split in pages, etc) here internally and only in the tests environment, because you're adding your own logic that might not be what the server is doing. Either do real api calls or read JSON files with real api responses content.
I'll leave you a general comment detailing this.

Reply:
Class is gone 👍